subprojects/core-api/src/main/java/org/gradle/internal/component/local/model/ComponentFileArtifactIdentifierWithOriginal.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.component.local.model;
+
+import org.gradle.api.artifacts.component.ComponentArtifactIdentifier;
+
+/**
+ * A component file artifact identifier that tracks the original file name.
+ *
+ * TODO: Rename to TransformedComponentFileArtifactIdentifier and
+ *  rename TransformedComponentFileArtifactIdentifier to DefaultTransformedComponentFileArtifactIdentifier
+ */
+public interface ComponentFileArtifactIdentifierWithOriginal extends ComponentArtifactIdentifier {
```

Comment:
The reason is, that TransformedComponentFileArtifactIdentifier is in the dependency-management, so not available in the core project. :/

Reply:
Can we just move `TransformedComponentFileArtifactIdentifier` to core then?

Reply:
This points out some pretty bad dependency cycles here. This code _should_ depend on dependency-management, (since it is obviously using code from it). 

Not saying that should be addressed here but this points to some more widespread architectural issues

Reply:
We can also do that if that is fine with you

Reply:
Yeah lets avoid this interface

Reply:
Moved `TransformedComponentFileArtifactIdentifier ` to the core project and removed the interface